telemetry/telemetry_record.py

```diff
@@ -96,9 +156,356 @@ def failed_url(url, output):
+                                                trial_name: cold_time_data
+                                            }
+                                        }
+    # Write back to aggregate
+    pickle.dump(curr_aggregate_data, open(aggregate_path, 'wb'))
+
+def reset_old_files():
+    """Deletes old files to make way for a new run
+
+    Removes:
+    /home/jamshed/src/tools/perf/page_sets/url*
+    /home/jamshed/src/tools/perf/page_sets/data/url*
+    /home/jamshed/page_load_time/telemetry/data/wpr_source/*
+    """
+    commands = [
+        'rm -f /home/jamshed/src/tools/perf/page_sets/url*',
+        'rm -f /home/jamshed/src/tools/perf/page_sets/data/url*',
+        'rm -f /home/jamshed/page_load_time/telemetry/data/wpr_source/*',
+        'rm -f /home/jamshed/page_load_time/telemetry/temp/aggregate.db',
+        'rm -f '
+        '/home/jamshed/page_load_time/telemetry/temp/benchmark_results.db',
+        'rm -f /home/jamshed/page_load_time/telemetry/data/results.db',
+        'rm -f /home/jamshed/page_load_time/data/har/*',
+            ]
+
+    for cmd in commands:
+        p = Popen(cmd, shell=True)
+        p.wait()
+
+def trial_reset():
+    """Deletes files between trials
+
+    Removes:
+    /home/jamshed/page_load_time/telemetry/benchmark_results.db
+    """
+    commands = [
+        'rm -f '
+        '/home/jamshed/page_load_time/telemetry/temp/benchmark_results.db'
+        ]
+
+    for cmd in commands:
+        p = Popen(cmd, shell=True)
+        p.wait()
+
+def get_min_results():
+    """Converts temp/aggregate.db into results.db, only taking min times
+
+    Format of output:
+    {'http://jamnoise.com':
+        {'cold_time': 314.159,
+         'modified_cold_time': '300.201'},
+         ...
+    }
+    """
+    aggregate_path = \
+            '/home/jamshed/page_load_time/telemetry/temp/aggregate.db'
+    aggregate_data = {}
+    try:
+        aggregate_data = pickle.load(open(aggregate_path, 'rb'))
+    except IOError:
+        raise IOError('Could not read from {0}'.format(aggregate_path))
+
+    min_data = {}
+
+    for url in aggregate_data:
+        url_min_results = {}
+        url_results = aggregate_data[url]
+        cold_time_min = None
+        modified_time_min = None
+        if 'cold_times' in url_results:
+            for trial in url_results['cold_times']:
+                if type(url_results['cold_times'][trial]) == float or \
+                        type(url_results['cold_times'][trial]) == int:
+                            url_results['cold_times'][trial] = \
+                            [url_results['cold_times'][trial]]
+                if cold_time_min is None:
+                    cold_time_min = min(url_results['cold_times'][trial])
+                else:
+                    cold_time_min = min(cold_time_min,
+                            min(url_results['cold_times'][trial]))
+        if 'modified_cold_times' in url_results:
+            for trial in url_results['modified_cold_times']:
```

Comment:
Ah yes. I can just write a single file for each url using the base64 and append the trial number to the file, then aggregate them all at the end? Does this sound good?